### ISSUE TITLE
If Item Code not exists getting selling cost center fails on invoice creation

### ISSUE THREAD

**Issue by maxtorete** [frappe/erpnext#10914]
Hi all,

If you have an item without code in a sales order, the invoice creation will fail as it try to pull the selling cost center from that item. I don't know if it is better to skip the selling cost center pulling or maybe it could be linked in sales order item so it can be defined manually in Sales Order items table.

This is the traceback:

```
Request Error
Traceback (most recent call last):
  File "/home/intranet/frappe-bench/apps/frappe/frappe/app.py", line 57, in application
    response = frappe.handler.handle()
  File "/home/intranet/frappe-bench/apps/frappe/frappe/handler.py", line 22, in handle
    data = execute_cmd(cmd)
  File "/home/intranet/frappe-bench/apps/frappe/frappe/handler.py", line 53, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "/home/intranet/frappe-bench/apps/frappe/frappe/__init__.py", line 923, in call
    return fn(*args, **newargs)
  File "/home/intranet/frappe-bench/apps/frappe/frappe/model/mapper.py", line 27, in make_mapped_doc
    return method(source_name)
  File "/home/intranet/frappe-bench/apps/erpnext/erpnext/selling/doctype/sales_order/sales_order.py", line 570, in make_sales_invoice
    }, target_doc, postprocess, ignore_permissions=ignore_permissions)
  File "/home/intranet/frappe-bench/apps/frappe/frappe/model/mapper.py", line 107, in get_mapped_doc
    map_child_doc(source_d, target_doc, table_map, source_doc)
  File "/home/intranet/frappe-bench/apps/frappe/frappe/model/mapper.py", line 201, in map_child_doc
    map_doc(source_d, target_d, table_map, source_parent)
  File "/home/intranet/frappe-bench/apps/frappe/frappe/model/mapper.py", line 126, in map_doc
    table_map["postprocess"](source_doc, target_doc, source_parent)
  File "/home/intranet/frappe-bench/apps/erpnext/erpnext/selling/doctype/sales_order/sales_order.py", line 540, in update_item
    or item.selling_cost_center \
AttributeError: 'NoneType' object has no attribute 'selling_cost_center'
```

Kind regards,
Juan

**Comment by maxtorete** [comment]
I found a workaround as it first try to pull the selling cost center from a project if the sales order has one attached, so I "Make Project" from the sales order and it let me create the invoice.

**Comment by villiger** [comment]
I have this problem as well. I set `item_code` as not mandatory for `Sales Order Item`, so that I can enter free text for a row in `Sales Order`.

**Comment by villiger** [comment]
I created a pull quequest who fixes this #12656
